src/com/studentmanagement/dao/StudentDao.java

```diff
@@ -0,0 +1,207 @@
+        String query = "DELETE * FROM Student WHERE ID='"+id+"'";
+        Statement statement = getConnection().createStatement();
+        int resultSet = statement.executeUpdate(query);
+        if(resultSet != 0){
+            System.out.println("Deleted!!");
+            return true;
+        }
+        return false;
+    }
+    public List<Student> findByCities(String city1,String city2) throws SQLException{
+        String query ="SELECT * FROM Student WHERE city ='"+city1+"'"+" OR city ='"+city2+"'";
+        Statement statement = getConnection().createStatement();
+        ResultSet resultSet = statement.executeQuery(query);
+        List<Student> studentList = new ArrayList<>();
+        while (resultSet.next()){
+           Student student = new Student(resultSet);
+            studentList.add(student);
+        }
+        return studentList;
+    }
+
+    public boolean updateDob(Student student) throws SQLException{
+        String query = "UPDATE Student SET dob ='"+student.getDob()+"'"+ "WHERE ID="+student.getID();
+        Statement statement = getConnection().createStatement();
+        int resultSet = statement.executeUpdate(query);
+        if(resultSet != 0){
+            return true;
+        }
+        return false;
+    }
+    public boolean updateAllScore(double score) throws SQLException{
+        int length = findAll().size()+1;
+        String query = "UPDATE Student SET average_score ="+score+" WHERE ID BETWEEN 0 AND "+length;
+        Statement statement = getConnection().createStatement();
+        int resultSet = statement.executeUpdate(query);
+        if(resultSet !=0){
+            return true;
+        }
+        return false;
+    }
+    public int generateId() throws SQLException {
+        String query = "select max(id) as maxId from student ";
+        Statement stmt = getConnection().createStatement();
+        ResultSet resultSet = stmt.executeQuery(query);
+        if (resultSet.next()) {
+            return resultSet.getInt("maxId") + 1;
+        } else {
+            return 0;
+        }
+    }
+    public List<Student> searchByMajor(String major) throws SQLException{
+        String query = "SELECT * FROM Student WHERE major='"+major+"'";
+        Statement statement = getConnection().createStatement();
+        ResultSet resultSet = statement.executeQuery(query);
+        List<Student> studentList = new ArrayList<>();
+        if(resultSet.next()){
+            Student student = new Student();
+            studentList.add(student);
+        }
+        return studentList;
+    }
+
+    public boolean averageScoreStudent() throws  SQLException{
+        String query = "SELECT AVG(average_score) as AVERAGE from Student";
+        Statement statement = getConnection().createStatement();
+        ResultSet resultSet = statement.executeQuery(query);
+        if(resultSet.next()){
+            System.out.println("Average Score Is");
+            System.out.println(resultSet.getString(1));
```

Comment:
Ở đây e ko nên return trua/false mà là điểm trung bình của học sinh (Cái mà e query ra từ database). 
Sau đó ở bên service sẽ chịu trách nhiệm in điểm đó ra. 
Ví dụ ở đây là averageScore => trả về kiểu double.
Có 2 cách để e lấy ra và trả về giá trị:

String mediumScoreString = resultSet.getString(1); : Giống như cách e đã lấy
Sau đó convert sang double: double mediumScore = Double.parseDouble(mediumScoreString); Rồi return
Lấy double mediumScore = resultSet.getDouble("mediumScore"); Do ở câu query bên trên e có đặt tên cột
AVG(average_score) là mediumScore nên e có thể lấy trực tiếp giá trị này ra.